testing/README.md

```diff
@@ -0,0 +1,390 @@
+
+In addition to the testing framework, you generally need a test *runner*. Which runner to use will depend on the nature of your project:
+
+- For pure-JavaScript projects that do not need access to the DOM, avoid using a test runner and simply run the `mocha` CLI instead (if you need a simple DOM without actual rendering, you can also use [cheerio](https://github.com/cheeriojs/cheerio) to construct a fake DOM).
+
+- For Non-Rails projects that require a full DOM, use [Karma](https://karma-runner.github.io/1.0/index.html). It integrates well with other tools you are likely using, such as Webpack and code coverage tools.
+
+- Rails projects should use [Teaspoon](https://github.com/modeset/teaspoon), which provides an excellent GUI for running tests in the browser. If you are also using [sprockets-commoner](https://github.com/Shopify/sprockets-commoner) to transpile your JavaScript, make sure to include [teaspoon-bundle](https://github.com/Shopify/teaspoon-bundle) to generate your test bundle.
+
+There are a variety of other tools you might need depending on your project. Here are some additional recommendations we make:
+
+- **Test coverage**: we recommend using [nyc](https://github.com/istanbuljs/nyc).
+
+- **Rails fixtures**: if you are testing a component that has a non-trivial HTML component, using the actual partial improves the test’s resilience. We recommend using [Magic Lamp](https://github.com/crismali/magic_lamp) if you need access to real partials in a Rails application (if you are writing a JavaScript application, you should be able to import your partials using JavaScript imports).
+
+[↑ scrollTo('#table-of-contents')](#table-of-contents)
+
+
+
+## Style
+
+- Our [shared linting configuration](../packages/eslint-plugin-shopify) provides a mocha-/ sinon-/ chai-focused linting configuration which you can use to prevent many common mistakes. To use it, simply extend `plugin:shopify/mocha` in your ESLint configuration (full details can be found on that package’s README).
+
+- Use [TDD-style `assert` syntax](http://chaijs.com/api/assert/) over BDD-style `expect` or `should`.
+
+  > Why? This keeps JavaScript tests more in line with our style for writing Ruby tests, which helps everyone contribute to the test codebase more easily.
+
+  ```js
+  // bad
+  expect(2 === 1).to.be.false;
+  (2 === 1).should.be.false;
+
+  // good
+  assert.isFalse(2 === 1);
```

Comment:
If we want to drive home the fact that we want everyone to use Chai we should add in an import for assert because the first time I was skimming through this doc I totally thought this was Node's assert. Using `isFalse` may not be explicit enough to identify this is coming from Chai